### PR TITLE
CPLP-1927-api-apps-active-refactor

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1927

### CODE REVIEW

File: src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public AppsBusinessLogic(IPortalRepositories portalRepositories, IOfferSubscript
                     app.ShortDescription ?? Constants.ErrorString,
                     app.VendorCompanyName,
                     app.LicenseText ?? Constants.ErrorString,
-                    app.ThumbnailUrl ?? Constants.ErrorString,
+                    app.LeadPictureId,
```
Comment (ntruchsess) [inline src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs:75]: as this IEnumerable<Guid> and not just a single Id please rename this property to 'LeadPictureIds'

File: src/marketplace/Apps.Service/ViewModels/AppData.cs [hunk 2]
```diff
@@ -28,13 +28,13 @@ namespace Org.Eclipse.TractusX.Portal.Backend.Apps.Service.ViewModels;
 /// <param name="ShortDescription">Short description.</param>
 /// <param name="Provider">Provider.</param>
 /// <param name="Price">Price.</param>
-/// <param name="LeadPictureUri">Lead pircture URI.</param>
+/// <param name="LeadPictureId">Lead pircture Id.</param>
```
Comment (ntruchsess) [inline src/marketplace/Apps.Service/ViewModels/AppData.cs:31]: as this IEnumerable and not just a single Id please rename this property to 'LeadPictureIds'

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs [hunk 3]
```diff
@@ -62,7 +62,7 @@ public interface IOfferRepository
     /// </summary>
     /// <param name="languageShortName">The optional language shortName</param>
     /// <returns>Returns a async enumerable of (Guid Id, string? Name, string VendorCompanyName, IEnumerable<string> UseCaseNames, string? ThumbnailUrl, string? ShortDescription, string? LicenseText)> GetAllActiveAppsAsync(string? languageShortName)</returns>
-    IAsyncEnumerable<(Guid Id, string? Name, string VendorCompanyName, IEnumerable<string> UseCaseNames, string? ThumbnailUrl, string? ShortDescription, string? LicenseText)> GetAllActiveAppsAsync(string? languageShortName);
+    IAsyncEnumerable<(Guid Id, string? Name, string VendorCompanyName, IEnumerable<string> UseCaseNames, IEnumerable<Guid> LeadPictureId, string? ShortDescription, string? LicenseText)> GetAllActiveAppsAsync(string? languageShortName);
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs:65]: as this IEnumerable and not just a single Id please rename this property to 'LeadPictureIds'

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 4]
```diff
@@ -77,15 +77,15 @@ public void AttachAndModifyOffer(Guid offerId, Action<Offer> setOptionalParamete
         _context.Remove(new Offer(offerId, null!, default, default)).Entity;
 
     /// <inheritdoc />
-    public IAsyncEnumerable<(Guid Id, string? Name, string VendorCompanyName, IEnumerable<string> UseCaseNames, string? ThumbnailUrl, string? ShortDescription, string? LicenseText)> GetAllActiveAppsAsync(string? languageShortName) =>
+    public IAsyncEnumerable<(Guid Id, string? Name, string VendorCompanyName, IEnumerable<string> UseCaseNames, IEnumerable<Guid> LeadPictureId, string? ShortDescription, string? LicenseText)> GetAllActiveAppsAsync(string? languageShortName) =>
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:80]: as this IEnumerable and not just a single Id please rename this property to 'LeadPictureIds'

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 5]
```diff
@@ -77,15 +77,15 @@ public void AttachAndModifyOffer(Guid offerId, Action<Offer> setOptionalParamete
...
         _context.Offers.AsNoTracking()
             .Where(offer => offer.DateReleased.HasValue && offer.DateReleased <= DateTime.UtcNow && offer.OfferTypeId == OfferTypeId.APP && offer.OfferStatusId == OfferStatusId.ACTIVE)
-            .Select(a => new ValueTuple<Guid,string?,string,IEnumerable<string>,string?,string?,string?>(
+            .Select(a => new ValueTuple<Guid,string?,string,IEnumerable<string>,IEnumerable<Guid>,string?,string?>(
                 a.Id,
                 a.Name,
                 a.ProviderCompany!.Name, // This translates into a 'left join' which does return null for all columns if the foreingn key is null. The '!' just makes the compiler happy
                 a.UseCases.Select(uc => uc.Name),
-                a.ThumbnailUrl,
+                a.Documents.Where(document=>document.DocumentTypeId ==DocumentTypeId.APP_LEADIMAGE && document.DocumentStatusId != DocumentStatusId.INACTIVE).Select(document=>document.Id),
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:88]: formating. There should be a space in between == and DocumentTypeId.APP_LEADIMAGE

**Review by ntruchsess — DISMISSED** [review]

File: src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs]: fixed

File: src/marketplace/Apps.Service/ViewModels/AppData.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/ViewModels/AppData.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 4 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 5 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

File: src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs (hunk 1 above)
Comment (jjeroch) [reply src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs]: no its a single id; LeadPictureId is correct

File: src/marketplace/Apps.Service/ViewModels/AppData.cs (hunk 2 above)
Comment (jjeroch) [reply src/marketplace/Apps.Service/ViewModels/AppData.cs]: no its a single id; LeadPictureId is correct

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs (hunk 3 above)
Comment (jjeroch) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/IOfferRepository.cs]: no its a single id; LeadPictureId is correct

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 4 above)
Comment (jjeroch) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: no its a single id; LeadPictureId is correct

**Review by Phil91 — APPROVED** [review]

**Pull request merged**